Fixes crash during randomization of individual sound effects.

leggettc18 opened the pull request:
I was picking a random value between 2 (0 and 1 aren't useable sequences) and the size of the map. However I didn't realize that many of the keys for the sound effects have keys higher than the size of the map. Since it's an ordered map, I now instead grab the key off of the last entry in the map. This seems to work from my testing.

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-linux.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/481420972.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/481420973.zip)
  - [soh-switch.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/481420974.zip)
  - [soh-wiiu.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/481420975.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/481420976.zip)
<!--- section:artifacts:end -->

leggettc18 committed:
Fixes randomization of individual sound effects.

leggettc18 commented:
I didn't crash in my earlier testing but just crashed again now (or technically not a crash it's more like a softlock). So this is not ready after all.

leggettc18 committed:
Re-adds the mistakenly removed const specifier.

dcvz commented on soh/soh/Enhancements/sfx-editor/SfxEditor.cpp:
```diff
@@ -300,7 +300,7 @@ void Draw_SfxTab(const std::string& tabId, const std::map<u16, std::tuple<std::s
+                value = Random(2, map.end()->first);
                 if (map.contains(value)) {
                     auto [name, sfxKey, seqType] = map.at(value);
                     if (seqType & type) {
```
Following up further on discord suggestion:
```c++
while (true) {
    auto it = map.begin();
    std::advance(it, rand() % map.size());
    const auto& [value, seqData] = *it;
    const auto& [name, sfxKey, seqType] = seqData;
    if (seqType & type) {
        CVar_SetS32(cvarKey.c_str(), value);
        SohImGui::RequestCvarSaveOnNextTick();
        break;
    }
}
```

leggettc18 committed:
Implements new randomization syntax that actually fixes the crash.

leggettc18 committed:
Increases default width of sfx editor menu.

The default width was cutting off a lot of the text in the dropdowns due to the added Randomize button.

leggettc18 commented:
Thanks to dcvz's suggestion this should be ready for merge now. Also increased the default width of the sfx editor menu while I was here, since the default width cut off a lot of the text in the dropdowns when I added the Randomize button. Figured that was fine for Bradley Charlie since it's kind of a UX bug.
